src/routes/safes/safes.service.ts

```diff
@@ -206,12 +206,19 @@ export class SafesService {
     transactions: Array<MultisigTransaction>;
     walletAddress: string;
   }): number {
-    return args.transactions.reduce((acc, { confirmations }) => {
-      const isConfirmed = !!confirmations?.some((confirmation) => {
-        return confirmation.owner === args.walletAddress;
-      });
-      return isConfirmed ? acc - 1 : acc;
-    }, args.transactions.length);
+    return args.transactions.reduce(
+      (acc, { confirmationsRequired, confirmations }) => {
+        const isConfirmed = confirmationsRequired === 0;
+        const isSignedByWallet = confirmations?.some((confirmation) => {
```

Comment:
A slight optimization

```suggestion
        const isSignedByWallet = isConfirmed && confirmations?.some((confirmation) => {
```

Reply:
Added in 1dcf67e02f2fa9d046f63154ba84c64688319f71.